pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -95,38 +102,74 @@ def __init__(self, time, pqs):
         if self._pqs.shape[1] != 7:
             raise ValueError("`pqs` matrix shall have 7 columns.")
 
+        self._min_time = min(self.time)
+        self._max_time = max(self.time)
+
     def as_matrix(self, query_time):
+        """Get transformation matrix at given time.
+
+        Parameters
+        ----------
+        query_time : Union[float, array-like shape (...)]
+            Query time
+
+        Returns
+        -------
+        A2B_t : array, shape (4, 4) or (..., 4, 4)
+            Homogeneous transformation matrix / matrices at given time / times.
+        """
         pq = self._interpolate_pq_using_sclerp(query_time)
-        return transform_from_pq(pq)
+        transforms = transforms_from_pqs(pq)
+        transforms = np.squeeze(transforms)  # to keep the external API
+        return transforms
 
     def check_transforms(self):
         self._pqs[:, 3:] = norm_vectors(self._pqs[:, 3:])
         return self
 
     def _interpolate_pq_using_sclerp(self, query_time):
-        # identify the index of the preceding sample
-        idx_timestep_earlier_wrt_query_time = np.argmax(
-            self.time >= query_time) - 1
+        query_time_arr = np.atleast_1d(query_time)
 
-        # deal with first timestamp
-        idx_timestep_earlier_wrt_query_time = max(
-            idx_timestep_earlier_wrt_query_time, 0)
+        # identify the index of the preceding sample
+        idxs_timestep_earlier_wrt_query_time = np.searchsorted(
+            self.time, query_time_arr, side='right'
+        ) - 1
+
+        # deal with first and last timestamp
+        min_index = 0
+        max_index = self.time.shape[0] - 2
+        idxs_timestep_earlier_wrt_query_time = np.clip(
+            idxs_timestep_earlier_wrt_query_time, min_index, max_index)
+        idxs_timestep_later_wrt_query_time = \
+            idxs_timestep_earlier_wrt_query_time + 1
+        before_start = query_time_arr <= self._min_time
+        idxs_timestep_later_wrt_query_time[
+            before_start] = idxs_timestep_earlier_wrt_query_time[before_start]
+        after_end = query_time_arr >= self._max_time
+        idxs_timestep_earlier_wrt_query_time[
+            after_end] = idxs_timestep_later_wrt_query_time[after_end]
 
         # dual quaternion from preceding sample
-        t_prev = self.time[idx_timestep_earlier_wrt_query_time]
-        pq_prev = self._pqs[idx_timestep_earlier_wrt_query_time, :]
-        dq_prev = dual_quaternion_from_pq(pq_prev)
+        t_prev = self.time[idxs_timestep_earlier_wrt_query_time]
+        pq_prev = self._pqs[idxs_timestep_earlier_wrt_query_time]
+        dq_prev = dual_quaternions_from_pqs(pq_prev)
```

Comment:
What do you think about more "array-like" names, e.g.
```suggestion
        times_prev = self.time[idxs_timestep_earlier_wrt_query_time]
        pqs_prev = self._pqs[idxs_timestep_earlier_wrt_query_time]
        dqs_prev = dual_quaternions_from_pqs(pq_prev)
```

Same for `_next` dual quaternions ...